core/src/test/java/org/infinispan/test/op/TestFunctionalWriteOperation.java

```diff
@@ -0,0 +1,175 @@
+package org.infinispan.test.op;
+
+import static org.infinispan.container.versioning.InequalVersionComparisonResult.EQUAL;
+import static org.infinispan.functional.FunctionalTestUtils.rw;
+
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.commands.ReplicableCommand;
+import org.infinispan.commands.VisitableCommand;
+import org.infinispan.commands.functional.ReadWriteKeyCommand;
+import org.infinispan.commands.functional.ReadWriteKeyValueCommand;
+import org.infinispan.commands.triangle.BackupWriteCommand;
+import org.infinispan.commands.write.ValueMatcher;
+import org.infinispan.container.versioning.NumericVersion;
+import org.infinispan.context.Flag;
+import org.infinispan.functional.FunctionalMap;
+import org.infinispan.functional.FunctionalTestUtils;
+import org.infinispan.functional.MetaParam;
+import org.infinispan.functional.decorators.FunctionalAdvancedCache;
+
+/**
+* Represents a functional write operation to test.
+*
+* @author Dan Berindei
+* @since 12.0
```

Comment:
12.1

Reply:
@ryanemerson if I backport it to 12.0.1, then wouldn't the `@since 12.0` tag be correct? :)

Reply:
touche! :laughing: